pkg/virt-controller/watch/export/export.go

```diff
@@ -384,19 +393,143 @@ func (ctrl *VMExportController) handleIsSourcePvc(vmExport *exportv1.VirtualMach
 			}
 		}
 	} else {
-		if pod.Status.Phase == corev1.PodSucceeded || pod.Status.Phase == corev1.PodFailed {
-			// The server died or completed, delete the pod.
-			ctrl.Recorder.Eventf(vmExport, corev1.EventTypeWarning, exporterPodFailedOrCompletedEvent, "Exporter pod %s/%s succeeded or failed", pod.Namespace, pod.Name)
-			if err := ctrl.Client.CoreV1().Pods(vmExport.Namespace).Delete(context.Background(), pod.Name, metav1.DeleteOptions{}); err != nil {
+		if err := ctrl.handlePodSucceededOrFailed(vmExport, pod); err != nil {
+			return retry, err
+		}
+	}
+
+	return ctrl.updateVMExportPvcStatus(vmExport, pvcs, pod, service, inUse)
+
+}
+
+func (ctrl *VMExportController) handleIsVmSnapshot(vmExport *exportv1.VirtualMachineExport, service *corev1.Service) (time.Duration, error) {
+	var retry time.Duration
+	pvcs := make([]*corev1.PersistentVolumeClaim, 0)
+
+	vmSnapshot, exists, err := ctrl.getVmSnapshot(vmExport.Namespace, vmExport.Spec.Source.Name)
+	restoreableSnapshots := 0
+	if err != nil {
+		return 0, err
+	} else if exists {
+		pvcs, restoreableSnapshots, err = ctrl.handlePVCsForVirtualMachineSnapshot(vmExport, vmSnapshot)
```

Comment:
You should check that the snapshot is "ready to use" before doing anything with it.  And if it is not ready, you either have to requeue or watch vmsnapshots

Reply:
Done